File: backend/app/schemas/email.py
from pydantic import BaseModel, EmailStr
from datetime import datetime
from typing import List, Optional

class EmailPreview(BaseModel):
    id: str
    sender: str
    subject: str
    snippet: str
    date: datetime
    unread: bool

class SendEmailRequest(BaseModel):
    to: List[EmailStr]
    subject: str
    body: str

class EmailDetail(BaseModel):
    id: str
    sender: str
    subject: str
    date: datetime
    body: str
    dataset: str # 'inbox' or 'sent' etc, metadata if needed
    unread: bool

class PaginatedEmails(BaseModel):
    messages: List[EmailPreview]
    nextPageToken: Optional[str] = None

class ReplyEmailRequest(BaseModel):
    body: str

class ForwardEmailRequest(BaseModel):
    to: List[EmailStr]
    body: str

File: frontend/src/types/email.ts
export interface EmailPreview {
    id: string;
    sender: string;
    subject: string;
    snippet: string;
    date: string;
    unread: boolean;
}

export interface EmailDetail extends EmailPreview {
    body: string;
}

export interface SendEmailPayload {
    to: string[];
    subject: string;
    body: string;
}

export interface PaginatedResponse {
    messages: EmailPreview[];
    nextPageToken: string | null;
}

export interface ReplyEmailPayload {
    body: string;
}

export interface ForwardEmailPayload {
    to: string[];
    body: string;
}
